### PR TITLE
Add signInAnonymously to auth-next

### PR DESCRIPTION
**Pull request by avolkovi**

### CODE REVIEW

File: packages-exp/auth-exp/src/core/providers/anonymous.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import { AuthCredential } from '../../model/auth_credential';
+import { IdTokenResponse } from '../../model/id_token';
+import { debugFail } from '../util/assert';
+
+export class AnonymousCredential implements AuthCredential {
+  providerId = ProviderId.ANONYMOUS;
+  signInMethod = SignInMethod.ANONYMOUS;
+
+  toJSON(): never {
+    debugFail('Method not implemented.');
```
Comment (avolkovi) [inline packages-exp/auth-exp/src/core/providers/anonymous.ts:30]: @samhorlbeck this is a fun way to handle unimplemented methods

Comment (samhorlbeck) [reply packages-exp/auth-exp/src/core/providers/anonymous.ts:29]: yeah that's clever

File: packages-exp/auth-exp/src/core/user/user_impl.ts [hunk 2]
```diff
@@ -65,7 +66,7 @@ export class UserImpl implements User {
   email: string | null;
   phoneNumber: string | null;
   photoURL: string | null;
-  isAnonymous = false;
+  isAnonymous: boolean = false;
```
Comment (samhorlbeck) [inline packages-exp/auth-exp/src/core/user/user_impl.ts:69]: Don't need to add the `:boolean`; go/tsstyle#type-inference

File: packages-exp/auth-exp/src/core/providers/anonymous.test.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ProviderId, SignInMethod } from '@firebase/auth-types-exp';
+import { expect, use } from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+import { AnonymousCredential, AnonymousProvider } from './anonymous';
+import { mockAuth } from '../../../test/mock_auth';
```
Comment (samhorlbeck) [inline packages-exp/auth-exp/src/core/providers/anonymous.test.ts:22]: This shouldn't work anymore. Each test needs to initialize its own auth object in a `beforeEach` block

File: packages-exp/auth-exp/src/core/strategies/anonymous.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ * limitations under the License.
+ */
+
+import * as externs from '@firebase/auth-types-exp';
+import { Auth } from '../../model/auth';
+import { AnonymousProvider } from '../providers/anonymous';
+import { UserCredentialImpl } from '../user/user_credential_impl';
+import { signInWithCredential } from './credential';
+
+export async function signInAnonymously(externAuth: externs.Auth): Promise<externs.UserCredential> {
```
Comment (samhorlbeck) [inline packages-exp/auth-exp/src/core/strategies/anonymous.ts:24]: Please add this as a re-export in src/index.ts

**Pull request merged**